src/parser/jobs/rpr/modules/Gauge.tsx

```diff
@@ -0,0 +1,178 @@
+ */
+const SOUL_GENERATORS_CDS: ActionKey[] = [
+	'SOUL_SLICE',
+	'SOUL_SCYTHE',
+]
+
+const SOUL_CONSUMERS: ActionKey[] = [
+	'BLOOD_STALK',
+	'GRIM_SWATHE',
+	'GLUTTONY',
+	'UNVEILED_GIBBET',
+	'UNVEILED_GALLOWS',
+]
+
+const SHROUD_GENERATORS: ActionKey[] = [
+	'GIBBET',
+	'GALLOWS',
+	'GUILLOTINE',
+]
+
+const SHROUD_CONSUMERS: ActionKey[] = [
+	'ENSHROUD',
+]
+
+/**
+ * Base amount for the combo actions
+ */
+const BASE_GAUGE_GENERATION_AMOUNT = 10
+
+/**
+ * Base amount for the soul gauge consumption
+ */
+const BASE_GAUGE_CONSUMPTION_AMOUNT = 50
+const PLENTIFUL_HARVEST_SHROUD_GENERATION_AMOUNT = 50
+const ENSHROUD_GAUGE_CONSUMPTION = 50
+
+/**
+ * Amount generated for the "Soul" skills, which are cooldowns
+ */
+const SOUL_GAUGE_GENERATION_AMOUNT = 50
+
+export class Gauge extends CoreGauge {
+	@dependency private suggestions!: Suggestions
+
+	private soulGauge = this.add(new CounterGauge({
+		minimum: 0,
+		maximum: SOUL_MAX_VALUE,
+		graph: {
+			label: 'Soul Gauge',
+			color: SOUL_GAUGE_COLOR.fade(0.25),
+		},
+	}))
+
+	private shroudGauge = this.add(new CounterGauge({
+		minimum: 0,
+		maximum: SHROUD_MAX_VALUE,
+		graph: {
+			label: 'Shroud Gauge',
+			color: SHROUD_GAUGE_COLOR.fade(0.25),
+		},
+	}))
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		// soul gauge
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_GENERATORS)), this.onSoulGeneration)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_GENERATORS_CDS)), this.onSoulGenerationCds)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_CONSUMERS)), this.onSoulConsumption)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.PLENTIFUL_HARVEST.id), this.onPlentifulHarvest)
+
+		// shroud gauge
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SHROUD_GENERATORS)), this.onShroudGeneration)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SHROUD_CONSUMERS)), this.onShroudConsumption)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.ENSHROUD.id), this.onEnshroud)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onSoulGeneration() {
+		this.soulGauge.generate(BASE_GAUGE_GENERATION_AMOUNT)
+	}
+
+	private onSoulGenerationCds() {
+		this.soulGauge.generate(SOUL_GAUGE_GENERATION_AMOUNT)
+	}
+
+	private onSoulConsumption() {
+		this.soulGauge.spend(BASE_GAUGE_CONSUMPTION_AMOUNT)
+	}
+
+	private onShroudGeneration() {
+		this.shroudGauge.generate(BASE_GAUGE_GENERATION_AMOUNT)
+	}
+
+	private onPlentifulHarvest() {
+		this.shroudGauge.generate(PLENTIFUL_HARVEST_SHROUD_GENERATION_AMOUNT)
+	}
+
+	private onEnshroud() {
+		this.shroudGauge.spend(ENSHROUD_GAUGE_CONSUMPTION)
+	}
+
+	private onShroudConsumption() {
+		this.shroudGauge.spend(BASE_GAUGE_CONSUMPTION_AMOUNT)
+	}
```

Comment:
These need to be restructured. Your base generation is the same, so realistically a switch for the +50 skills and defaulting to the +10 is fine. Consumers are all 50 as well, so they can effectively be a single hook using the action to determine which gauge.